test/e2e/util.go

```diff
@@ -571,6 +536,68 @@ func RunRC(c *client.Client, name string, ns, image string, replicas int) error
 	return nil
 }
 
+func ResizeRC(c *client.Client, ns, name string, size int) error {
+	By(fmt.Sprintf("Resizing replication controller %s to %d", name, size))
+	rc, err := c.ReplicationControllers(ns).Get(name)
+	if err != nil {
+		return fmt.Errorf("Failed to find replication controller %s in namespace %s: %v", name, ns, err)
+	}
+	rc.Spec.Replicas = size
+	if _, err := c.ReplicationControllers(ns).Update(rc); err != nil {
+		return fmt.Errorf("Failed to resize replication controller %s: %v", name, err)
+	}
+
+	// Wait up to 10 minutes until all replicas are killed.
+	endTime := time.Now().Add(time.Minute * 10)
+	for {
+		if time.Now().After(endTime) {
+			return fmt.Errorf("Timeout while waiting for replication controller %s replicas to %s", name, size)
+		}
+		remainingTime := endTime.Sub(time.Now())
+		err := wait.Poll(time.Second, remainingTime, client.ControllerHasDesiredReplicas(c, rc))
+		if err != nil {
+			Logf("Error while waiting for replication controller %s replicas to read %v: %s", name, size, err)
+		} else {
+			break
+		}
+	}
+
+	// Wait up to 10 minutes for pods to become Running.
+	running := false
+	label := labels.SelectorFromSet(labels.Set(map[string]string{"name": name}))
+	for start := time.Now(); time.Since(start) < 10*time.Minute; time.Sleep(5 * time.Second) {
```

Comment:
There is the same pattern in RunRC function. Since nobody complains let's leave it as it is now. When we decide to get rid of listing pods from e2e tests we will do it in both places.